### DISCUSSION TITLE
Random characters appear in hover

### DISCUSSION THREAD

**Discussion by creasty** [neoclide/coc.nvim#3180] [Q&A] [answered]
When I do hover, random numbers and symbols get sometimes rendered in the first column.
It seems like it happens in typescript where the type is inferred and/or is complex enough to be described in multiple lines.

![Screen Shot 2021-06-29 at 14 26 06](https://user-images.githubusercontent.com/1695538/123742060-0d2c0b00-d8e6-11eb-95d3-e3a90cede5c3.png)

Configurations:
https://github.com/creasty/dotfiles/blob/master/vim/coc-settings.json
https://github.com/creasty/dotfiles/blob/master/vim/autoload/vimrc/plugin/coc.vim

**Comment by chemzqm** [comment, marked as answer]
Seems bug with your vim.

**Reply by creasty** [reply]
@chemzqm Hmm...
Do you happen to have any idea what might cause this?...

I'd also appreciated it if you could point me to the location of the code where the first columns are allocated/rendered. I'm assuming they're for simulating padding, but couldn't figure out how exactly this space is created.
Regarding the float, is there any way I can inspect what is shown in a currently opening float?

----

FYI, I'm using neovim v0.4.4.

```
$ nvim -v
NVIM v0.4.4
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20210318-48132-k7r45c/neovim-0.4.4/build/config -I/tmp/neovim-20210318-48132-k7r45c/neovim-0.4.4/src -I/usr/local/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/usr/local/opt/gettext/include -I/tmp/neovim-20210318-48132-k7r45c/neovim-0.4.4/build/src/nvim/auto -I/tmp/neovim-20210318-48132-k7r45c/neovim-0.4.4/build/include
Compiled by brew@BigSur

Features: +acl +iconv +tui
See ":help feature-compile"
```

**Reply by chemzqm** [reply]
No idea, try minimal vimrc.

**Comment by creasty** [comment]
Finally, found the root cause.

coc.nvim uses `foldcolumn` to add padding to the right side of a window.
https://github.com/neoclide/coc.nvim/blob/34d9d52f4e030a7f1c5e15afa1928cad6d9d903d/autoload/coc/float.vim#L296

There is `set foldmethod=indent` in my config which enables folds for every indent, resulting in showing characters that are seemingly random but actually indicate the state of folds (like open or close, or nest level).
https://github.com/creasty/dotfiles/blob/cb280d39f79d63469e880c2b9ee4c3b6f1517f49/vim/plugin/vimrc/appearance.vim#L85-L90

Although I think it's reasonable to include this setting in `coc#float#nvim_right_pad`, I've worked around the issue by disabling folds in coc float windows.
https://github.com/creasty/dotfiles/commit/5b05651724a9b508a1497061e56cbb6ae91312cf

![Screen Shot 2021-06-29 at 21 38 41](https://user-images.githubusercontent.com/1695538/123798690-6a45b200-d922-11eb-975f-0ca878cae860.png)

I'll throw a PR if you find this fix plausible.
